src/Plugin/rest/resource/WebformAllFormSubmissions.php

```diff
@@ -106,53 +109,91 @@ public function get(string $webform_id): ModifiedResourceResponse {
         ],
       ];
 
-      return new ModifiedResourceResponse($errors, 401);
+      return new ModifiedResourceResponse($errors, Response::HTTP_UNAUTHORIZED);
     }
 
     $result = ['webform_id' => $webform_id];
 
+    try {
+      $submissionEntityStorage = $this->entityTypeManager->getStorage('webform_submission');
+    }
+    catch (InvalidPluginDefinitionException | PluginNotFoundException $e) {
+      $errors = [
+        'error' => [
+          'message' => $this->t('Could not load webform submission storage'),
+        ],
+      ];
+
+      return new ModifiedResourceResponse($errors, Response::HTTP_INTERNAL_SERVER_ERROR);
+    }
+
     // Query for webform submissions with this webform_id.
-    $query = $this->entityTypeManager->getStorage('webform_submission')->getQuery()
+    $submissionQuery = $submissionEntityStorage->getQuery()
       ->condition('webform_id', $webform_id);
 
-    $startTimestamp = $this->currentRequest->query->get('starttime');
-    if (is_numeric($startTimestamp)) {
-      $query->condition('created', $startTimestamp, '>=');
-      $result['starttime'] = $startTimestamp;
-    }
+    foreach (self::ALLOWED_DATETIME_QUERY_PARAMS as $param => $operator) {
+      $errors = $this->updateSubmissionQuery($this->currentRequest, $submissionQuery, $param, $operator, $result);
```

Comment:
Inline this function call.

Reply:
Removed `updateSubmissionQuery` and moved its logic into for loop.